### ISSUE TITLE
applySchema does not accept plain functions

### ISSUE THREAD

**Issue by danielweinmann** [seasonedcc/composable-functions#163]
I get the following error when trying to use `applySchema` with a plain function:

```ts
  const increment = (value: number) => value + 1
  const incrementWithSchema = applySchema(z.number())(increment)

// Argument of type '(value: number) => number' is not assignable to parameter of type '((input: number, context: unknown) => Promise<Result<unknown>>) & { kind: "composable"; }'.
//   Type '(value: number) => number' is not assignable to type '(input: number, context: unknown) => Promise<Result<unknown>>'.
//     Type 'number' is not assignable to type 'Promise<Result<unknown>>'.ts(2345)
```

If I wrap it with `composable`, everything works fine.

```ts
  const increment = composable((value: number) => value + 1)
  const incrementWithSchema = applySchema(z.number())(increment)
```

**Comment by gustavoguichard** [comment]
Yup, this is planned. And, once we do that, `applySchema` and `withSchema` will do the same thing. Use withSchema for plain functions in the meantime

**Comment by gustavoguichard** [comment]
@danielweinmann #164 should solve this issue =)

**Comment by gustavoguichard** [comment]
Solved on v4.3
